From 628cf1461dadd6814baafa5d98861d74c16a9df4 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <aisheng.dong@nxp.com>
Date: Wed, 6 Sep 2017 21:47:14 +0800
Subject: [PATCH 3313/5242] MLK-17491-13 dts: imx7ulp: improve the gpio using

commit  b473c6d5a3bef9a16778033a3cfeb584c0f31ec1 from
https://source.codeaurora.org/external/imx/linux-imx.git

Currently people have no idea on which pad is correspding to which gpio
controller as there's no hints in dts. Let's add a proper prefix for gpio
nodes as follows in dts to make it much easier to use.
gpio0 = &gpio_ptc;
gpio1 = &gpio_ptd;
gpio2 = &gpio_pte;
gpio3 = &gpio_ptf;

Reviewed-by: Fugang Duan <fugang.duan@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk-sd1.dts    |    6 +++---
 arch/arm/boot/dts/imx7ulp-evk-wm8960.dts |    4 ++--
 arch/arm/boot/dts/imx7ulp-evk.dts        |   18 +++++++++---------
 arch/arm/boot/dts/imx7ulp.dtsi           |   16 ++++++++--------
 4 files changed, 22 insertions(+), 22 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk-sd1.dts b/arch/arm/boot/dts/imx7ulp-evk-sd1.dts
index 4c03338..9936183 100644
--- a/arch/arm/boot/dts/imx7ulp-evk-sd1.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk-sd1.dts
@@ -17,7 +17,7 @@
 			regulator-name = "VSD_3V3B";
 			regulator-min-microvolt = <3300000>;
 			regulator-max-microvolt = <3300000>;
-			gpio = <&gpio2 11 GPIO_ACTIVE_HIGH>;
+			gpio = <&gpio_pte 11 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
 	};
@@ -33,8 +33,8 @@
 	pinctrl-1 = <&pinctrl_usdhc1 &pinctrl_usdhc1_rst>;
 	pinctrl-2 = <&pinctrl_usdhc1 &pinctrl_usdhc1_rst>;
 	pinctrl-3 = <&pinctrl_usdhc1 &pinctrl_usdhc1_rst>;
-	cd-gpios = <&gpio2 13 GPIO_ACTIVE_LOW>;
-	wp-gpios = <&gpio2 12 GPIO_ACTIVE_HIGH>;
+	cd-gpios = <&gpio_pte 13 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&gpio_pte 12 GPIO_ACTIVE_HIGH>;
 	vmmc-supply = <&reg_vsd_3v3b>;
 	/delete-property/non-removable;
 	/delete-property/cd-post;
diff --git a/arch/arm/boot/dts/imx7ulp-evk-wm8960.dts b/arch/arm/boot/dts/imx7ulp-evk-wm8960.dts
index 3389d95..25d431e 100644
--- a/arch/arm/boot/dts/imx7ulp-evk-wm8960.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk-wm8960.dts
@@ -125,8 +125,8 @@
 		codec-master;
 		/* JD3: hp detect high for headphone*/
 		hp-det = <3 0>;
-		hp-det-gpios = <&gpio3 0 0>;
-		mic-det-gpios = <&gpio3 0 0>;
+		hp-det-gpios = <&gpio_ptf 0 0>;
+		mic-det-gpios = <&gpio_ptf 0 0>;
 		audio-routing =
 			"Headphone Jack", "HP_L",
 			"Headphone Jack", "HP_R",
diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 27fa574..18b7301 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -40,7 +40,7 @@
 
 	mipi_dsi_reset: mipi-dsi-reset {
 		compatible = "gpio-reset";
-		reset-gpios = <&gpio0 19 GPIO_ACTIVE_LOW>;
+		reset-gpios = <&gpio_ptc 19 GPIO_ACTIVE_LOW>;
 		reset-delay-us = <1000>;
 		#reset-cells = <0>;
 	};
@@ -66,7 +66,7 @@
 			regulator-name = "usb_otg1_vbus";
 			regulator-min-microvolt = <5000000>;
 			regulator-max-microvolt = <5000000>;
-			gpio = <&gpio0 0 GPIO_ACTIVE_HIGH>;
+			gpio = <&gpio_ptc 0 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
 
@@ -76,7 +76,7 @@
 			regulator-name = "VSD_3V3";
 			regulator-min-microvolt = <3300000>;
 			regulator-max-microvolt = <3300000>;
-			gpio = <&gpio1 0 GPIO_ACTIVE_HIGH>;
+			gpio = <&gpio_ptd 0 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
 
@@ -84,7 +84,7 @@
 
 	extcon_usb1: extcon_usb1 {
 		compatible = "linux,extcon-usb-gpio";
-		id-gpio = <&gpio0 13 GPIO_ACTIVE_HIGH>;
+		id-gpio = <&gpio_ptc 13 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default", "sleep";
 		pinctrl-0 = <&pinctrl_extcon_usb1>;
 		pinctrl-1 = <&pinctrl_extcon_usb1>;
@@ -389,11 +389,11 @@
 	focaltech@38 {
 		compatible = "focaltech,fts";
 		reg = <0x38>;
-		interrupt-parent = <&gpio3>;
+		interrupt-parent = <&gpio_ptf>;
 		interrupts = <0 0x02>;
 		focaltech,panel-type = <FT5426>;
-		focaltech,reset-gpio = <&gpio3 1 0x01>;
-		focaltech,irq-gpio = <&gpio3 0 0x02>;
+		focaltech,reset-gpio = <&gpio_ptf 1 0x01>;
+		focaltech,irq-gpio = <&gpio_ptf 0 0x02>;
 		focaltech,max-touch-number = <5>;
 		focaltech,display-coords =  <0 0 272 480>;
 
@@ -419,7 +419,7 @@
 		reg = <0x3d>; /* PD pin is low */
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_dsi_hdmi>;
-		interrupt-parent = <&gpio0>;
+		interrupt-parent = <&gpio_ptc>;
 		interrupts = <18 IRQ_TYPE_EDGE_FALLING>;
 		video-mode = <1>; /*
 			           * Only support CEA modes.
@@ -514,7 +514,7 @@
 	pinctrl-1 = <&pinctrl_usdhc0>;
 	pinctrl-2 = <&pinctrl_usdhc0>;
 	pinctrl-3 = <&pinctrl_usdhc0>;
-	cd-gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+	cd-gpios = <&gpio_ptc 10 GPIO_ACTIVE_LOW>;
 	vmmc-supply = <&reg_vsd_3v3>;
 	vqmmc-supply = <&vldo2_reg>;
 	status = "okay";
diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index 121a1d4..38f57c9 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -18,10 +18,10 @@
 	interrupt-parent = <&intc>;
 
 	aliases {
-		gpio0 = &gpio0;
-		gpio1 = &gpio1;
-		gpio2 = &gpio2;
-		gpio3 = &gpio3;
+		gpio0 = &gpio_ptc;
+		gpio1 = &gpio_ptd;
+		gpio2 = &gpio_pte;
+		gpio3 = &gpio_ptf;
 		mmc0 = &usdhc0;
 		mmc1 = &usdhc1;
 		serial0 = &lpuart4;
@@ -522,7 +522,7 @@
 			reg = <0x40ac0000 0x1000>;
 		};
 
-		gpio0: gpio@40ae0000 {
+		gpio_ptc: gpio@40ae0000 {
 			compatible = "fsl,vf610-gpio";
 			reg = <0x40ae0000 0x1000 0x400F0000 0x40>;
 			gpio-controller;
@@ -533,7 +533,7 @@
 			gpio-ranges = <&iomuxc1 0 0 32>;
 		};
 
-		gpio1: gpio@40af0000 {
+		gpio_ptd: gpio@40af0000 {
 			compatible = "fsl,vf610-gpio";
 			reg = <0x40af0000 0x1000 0x400F0040 0x40>;
 			gpio-controller;
@@ -544,7 +544,7 @@
 			gpio-ranges = <&iomuxc1 0 32 32>;
 		};
 
-		gpio2: gpio@40b00000 {
+		gpio_pte: gpio@40b00000 {
 			compatible = "fsl,vf610-gpio";
 			reg = <0x40b00000 0x1000 0x400F0080 0x40>;
 			gpio-controller;
@@ -555,7 +555,7 @@
 			gpio-ranges = <&iomuxc1 0 64 32>;
 		};
 
-		gpio3: gpio@40b10000 {
+		gpio_ptf: gpio@40b10000 {
 			compatible = "fsl,vf610-gpio";
 			reg = <0x40b10000 0x1000 0x400F00c0 0x40>;
 			gpio-controller;
-- 
1.7.9.5

